### DISCUSSION TITLE
Now what?

### DISCUSSION THREAD

**Discussion by GerryDazoo** [GerryDazoo/Slinger#321] [General]
My question for the masses is now what?
Thanks to the herculian efforts by the community members we now have a procedure for un-bricking the ProHDs which is great. It's my personal opinion now that the Solos and Pros are unrecoverable with software.
So, as far as priorities go, I'd like to tackle being able to reconfigure the video source on the 350/500/M1/M2 that are still alive. We need this so replacement boxes off Ebay can be setup and used after a factory reset.
I've been a little out of the loop, can someone confirm that the SPM1.5 does not work with the 500's?
I don't mind writing some code or getting some help writing some code if we can figure out what needs to be done.
I do have some complete packet captures of a factory reset I did before the shutdown.

**Comment by tom76017** [comment]
I'd like to propose that you generate a new fork for future enhancements.  I like your idea about flexibility on changing source, but I also would like to see your current version of python stabilize.  My M1 is running perfectly.  I've been running it both internally and remotely for over a month.

There still seems to be some confusion for new users.  It would be nice to see an update to the setup/install process depending in the SB type and if a hard reset has been performed.  I'd be interested in seeing a github poll on the number of active users by SB model since EOL N9.

**Comment by lepmail** [comment]
I'd be more than happy to help examine the packets to extract whatever information we can.  Was that a 500 factory reset you captured?   I'm curious on why you don't think the Solos/Pros can be recovered with FW rollback, but I too agree that the "newer" boxes would/should have higher priority. I second Tom's suggestions above - most folks just want to be able to use Slinger, while geekier ones like me are willing to dig deeper looking for answers.  Let's get to the bottom of this! ;)

**Reply by GerryDazoo** [reply]
Well maybe the solo could be recovered with a FW rollback. The proplem is we can't login into the box to make that happen.

**Reply by GerryDazoo** [reply]
Captures are here. http://home.dazoo.net:64888/Slinger/
192.168.117.2 = PC running Desktop app
192.168.117.110 = IP address of 500
192.168.117.129 = IP for M1.
My immediate problem is I've been unable to generate the proper HTTP authentication header to get the slingbox to respond without an Auth failure. There was this thread on reddit, but the guy who said he knew how to do it has gone radio silent.
https://www.reddit.com/r/slingbox/comments/ygb695/im_making_a_plugandplay_replacement_for_the_sling/

**Reply by lepmail** [reply]
Gerry - Just done with daytime "work"... getting ready to start excavating. Thanks for the links

**Reply by gbjbasdw** [reply]
I totally agree with Lepmail when he says "most folks just want to be able to use Slinger". It's not that we don't want to spend time getting it working or are lazy - it is more that we don't have the expertise to get involved in the nuts and bolts.

It gets really confusing reading all the posts here and trying to understand where it is at. I realise that this is a technical forum but it would be nice to have a "latest status" somewhere. I saw there was a version 1.08e pdf but I couldn't seems to access it. Undoubtedly me being stupid !

**Reply by socelillyhammer** [reply]
reading the posts.  thank you guys for all the work youre doing.  id love to learn how to do what you guys do one day and understand it, but until then kudos and thanks

**Comment by RandomMan719** [comment]


**Reply by GerryDazoo** [reply]
In the video he never configured the Remote. We already knew what he did worked.

**Comment by robosmith** [comment]


**Reply by GerryDazoo** [reply]
No, not the right place I think there's a Tivo remote thread.

**Comment by BobC1954** [comment]
One thing I think would be good Gerry, is if a remote (with numerical entry) could be made to work for ProHD with Slinger. It's currently quite difficult to change channels using CH+ or -, when needing to step slowly through many channels to get to the one you want. I take it that it is quite difficult to do?

**Reply by GerryDazoo** [reply]
Bob, The ProHD remote has a place to directly tune to a channel. Does that not work in the UK?
email me directly

**Reply by BobC1954** [reply]
Email sent

**Comment by sp222222** [comment]
SPM 1.5 and spm 2.0 do NOT work with 500. 

I'd like to see someone tackle the jtag  for solo (out of my realm of knowledge). I think they can be recovered if we had admin. I've been ordering solos off ebay with hit and miss (I'll confirm the miss in a month or so when I can get to the remote solo setup)

**Comment by floormanager1** [comment]
I agree to priority the 350/M1/M2 route.  There's plenty on ebay and I would be happy to buy a spare but the configuration and password issues is preventing me at present.

**Reply by GerryDazoo** [reply]
What password issue are you referring to? We know the password after a factory reset is "admin"

**Reply by floormanager1** [reply]
For 350s, M1?  They have a long, random password don't they? Or is that added by Sing once they are configured (or used to be)?

**Reply by GerryDazoo** [reply]
Added by Sling.

**Reply by floormanager1** [reply]
I see, so I can get a M1 from ebay, reset it and the password will be admin.  Excellent (off to ebay now)

**Reply by GerryDazoo** [reply]
Yes we know the default password but we can't yet configure the inputs after a factory reset so the remote won't work.

**Comment by sp222222** [comment]
Wait? for a M1 ? I don’t think so. If it is, I’ll buy one too but there
isn’t a player to configure inputs.

**Comment by regmar505** [comment]
Thanks Gerry for your incredible work, my M2 streams to a remote server and it goes crazy well. I blocked in/out internet access to it and I wonder what it is going to happen if my M2 keeps calling home and nobody answers.  I did not see a clear answer on why some Slingbox models were wiped out and some not, it makes me worried. Should we power off the unit when unused until we have a clearer view?

**Reply by sp222222** [reply]
that M2 could care less about getting an answer. if your M2 works with slinGER , set it and forget it. you've already put rules to block. Be glad you didn't get bricked like most of us using the older boxes. And you haven't read enough if you didn't see pro, solo proHD, AV, tuner and original all get bricked within seconds if left exposed to outbound traffic. m1, m2, 500, 350 are safe for now from this but sure as heck won't hurt to block their outbound traffic either.

**Comment by Pthepeg** [comment]
I can't speak for the masses but could you make an API of sorts to return server status? I have my own remote html page working nicely.  However I don't get the packet count and number of streams info that shows up in the <H3> tag as when using the remote.txt

**Reply by Pthepeg** [reply]
Lol. The \<H3\> tag I meant to say

**Reply by GerryDazoo** [reply]
This is the string the code is looking for
`<h3>Status:%s</h3>`
sorry I didn't preview before posting. stick this in the body of your page where you want the status to be. Use this exact string.

**Reply by blasco403** [reply]
Hey Gerry,

I am using the HTML TiVo remote that I have edited for my STB's required codes.

I am not a programmer, I can edit and figure things out usually with examples, but don't write code.

How would I add this to the HTML file to display the status?

Thanks for any help/pointers someone can provide.

Ron.

**Reply by Toei79** [reply]
> Hey Gerry,
> 
> I am using the HTML TiVo remote that I have edited for my STB's required codes.
> 
> I am not a programmer, I can edit and figure things out usually with examples, but don't write code.
> 
> How would I add this to the HTML file to display the status?
> 
> Thanks for any help/pointers someone can provide.
> 
> Ron.

exactly my case,  im struggling a lot with all this stuff sometimes from linux sometimes take me months. because its not there examples, things are very straight forward but when you not touch it ,or enough familiar with it, its really hard.

examples help a lot.  its kinda like mathematics books sometimes. 

talking about this a good project for what i understand , can multiple streams like X10 watchers so you can share if you want, so far im not figure out yet.

https://github.com/segator/proxylive
sorry off topic.

**Reply by GerryDazoo** [reply]
Blasco: 
please see my previous now edited comment.

**Reply by blasco403** [reply]
Thanks Gerry,

Got it working.

**Comment by robosmith** [comment]
I would like to see (and help with) some work involving a more robust connection with the video player client.
When my Internet gets busy (around prime time) the connection with Slinger becomes somewhat unreliable.
Every 5-10 minutes, MxPlayer will drop the stream and require reinitialization. Not sure it's possible, but it would be nice if Slinger could detect that condition and fill the the gaps or automatically reinitialize the connection.

It seems like SlingPlayer apk would temporarily pixelate or skip frames under those conditions to keep the stream alive.

**Reply by GerryDazoo** [reply]
What your suggesting is impossible. Media players don't accept unsolicited streaming requests so there is no way for slinger to remake the connection when it drops. But this reminds of the famous quote. "Dr. it hurts when I do this. Dr. well dont' do that"
So if it's dropping when your internet is busy reduce your bandwidth setting and/or  configure your media player to restart a dropped connection automatically.:

**Reply by Toei79** [reply]
whats your upload speed? because if its limited then there its the issue maybe.

like 5 years ago i had comcast my upload speed was 5 mbps  cable, now i have fiber then its not issue.  thats important imo.

**Reply by GerryDazoo** [reply]
Another note. Because there is no feedback from the media player unlike the integrated SlingBox player there is no way for my code to know what you're seeing and adjust the stream accordingly. What I do is configure multiple slingbox configurations with different bandwidth setting and use the appropriate one when connecting. i.e. I've got one high bandwidth for LAN and another with lower bandwidth settings for viewing on my phone when connected via cellular data.

**Reply by flipper2222222** [reply]
configurations with different bandwidth setting and use the appropriate one when connecting. i.e. I've got one high bandwidth for LAN and another with lower bandwidth settings for viewing on my phone when connected via cellular data.

VideoBandwidth=2000  just wanted to confirm this 2000 figure means a 2MB upload speed from the slingbox location? thanks

**Reply by robosmith** [reply]
Ok, if there is no feedback from the video player about Internet overload/frame drops, then it would not be possible for Slinger to automatically down grade the video quality to a level which compensates for available bandwidth. @Toei79 upload speed at the source is not the issue; the issue is apparently the loading of the Internet over a 3000 mile link or the local loading.

**Reply by Toei79** [reply]
> Ok, if there is no feedback from the video player about Internet overload/frame drops, then it would not be possible for Slinger to automatically down grade the video quality to a level which compensates for available bandwidth. @Toei79 upload speed at the source is not the issue; the issue is apparently the loading of the Internet over a 3000 mile link or the local loading.

That sounds like servers proximity on Xbox 

Vpn tunnel can helps maybe?
I do with asus router, plus not port forwarding

**Reply by robosmith** [reply]
It really depends on where in the link there is insufficient capacity to keep the stream alive.

**Reply by robosmith** [reply]
@GerryDazoo is there a way (or could there be) to dynamically set the resolution of the video stream?
Maybe with a code sent by the Remote.

**Reply by sp222222** [reply]
lol that’s where slingbox made its bread and butter. it’s in the black box of technology. btw, for 3000 mile slingbox journey you need vpn with shorter hops which cost money on your side. something on each continent

**Reply by GerryDazoo** [reply]
A loong time ago in the 1.x days there was a button to change the resolution. But it didn't work very well because most players couldn't handle the dynamic video format changes and just dropoed the connection.  We see that behaviour these days on the proHD tuner when changing between SD and HD channels. And just changing the resolution didn't necessarily reduce the bandwidth being use and most likely you also want to adjust some of the other video parameters for best viewing. So this is what I do and recommend. Make a unified config with a number of slingbox sections for the same slingbox but with different video configurations and connect to the one that best fits your current situation. I've got three configs that I use, one high bandwidth and resolution for LAN, One lower resolution/bandwidth/frame rate for streaming over cellular data and an audio only for listening in the car.

**Reply by robosmith** [reply]
@GerryDazoo When the player starts dropping connections, I've noticed that the number of Slinger streams starts rising, as high as 8 streams. That seems to make re-establishing a connection very difficult. Re-booting Slinger helps a lot. Not sure why the number of streams grows under those conditions. Does Slinger shut down streams when they're disconnected?

**Reply by GerryDazoo** [reply]
You're using VLC aren't you. This is a VLC issue. He's the one not dropping the connections before starting a new one. Slinger does not close connections on it's own unless the video stream from the slingbox is interrupted for some reason.

**Comment by edutsilva** [comment]
Hi, once again, thank you @GerryDazoo for everything. And thank you all.
I will need to change the remote codes for my M2. Is it possible?
Thank you

**Reply by GerryDazoo** [reply]
I think so, I've had some success with the SlingPLayer Desktop V5.0.0.83

**Reply by edutsilva** [reply]
Do I have to reset it? Did you?
Mine was perfect until I changed the cable provider.
Thanx.

**Reply by GerryDazoo** [reply]
I don't think so but it couldn't hurt. How is it not perfect now? It did update the remote,yes?

**Reply by Toei79** [reply]
> I think so, I've had some success with the SlingPLayer Desktop V5.0.0.83

i have that version i cant setup there.

**Reply by edutsilva** [reply]
> I don't think so but it couldn't hurt. How is it not perfect now? It did update the remote,yes?

It´s not perfect, because the remote does not work anymore. I changed my cable provider and now I need to change the M2 remote control configuration. Is there a way? Which client version should I use?

**Reply by GerryDazoo** [reply]
SDT 5.0.0.83. Make sure where you're running the application has Internet access to the Sling servers.

**Reply by edutsilva** [reply]
Is there any sling server still up???
So, let me check if I get everything right to fix my M2:
1) Reset it
2) Using SDT 5.0.0.83 with internet access I will be able to access the M2 and configure it. Default password: admin

Thanx

**Reply by GerryDazoo** [reply]
If all you want to do is reconfigure a video input, the (factory) reset step should not needed. Just run the setup with the existing password.
